src/models/Product.js

```diff
@@ -1,23 +1,38 @@
 const mongoose = require('mongoose');
+const Review = require('./Review');
+const Logger = require('../core/Logger');
 
-const productSchema = new mongoose.Schema({
-    name: String,
-    price: Number,
-    desc: String,
-    imageUrl: String,
-    quantity: Number,
-    rating: Number,
-    reviews: [
-        {
+const productSchema = new mongoose.Schema(
+    {
+        name: String,
+        price: Number,
+        desc: String,
+        imageUrl: String,
+        quantity: Number,
+        rating: Number,
+        reviews: [
+            {
+                type: mongoose.Schema.Types.ObjectId,
+                ref: 'Review',
+            },
+        ],
+        author: {
             type: mongoose.Schema.Types.ObjectId,
-            ref: 'Review',
+            ref: 'User',
         },
-    ],
-    author: {
-        type: mongoose.Schema.Types.ObjectId,
-        ref: 'User',
     },
-}, { versionKey: false, timestamps: true });
+    { versionKey: false, timestamps: true },
+);
+
+productSchema.pre(
+    'deleteOne',
+    { document: true, query: false },
+    async function (next) {
+        Logger.info('inside deleteOne pre hook');
+        await Review.deleteMany({ _id: { $in: this.reviews } });
```

Comment:
We should not execute the query if there are not reviews.

```suggestion
      if(this.reviews.length){
              await Review.deleteMany({ _id: { $in: this.reviews } });
      }
```